tests/PhpOption/Tests/SomeTest.php

```diff
@@ -105,10 +105,21 @@ public function testFilterIsOneOf()
 
         $this->assertSame($some, $some->filterIsOneOf(\stdClass::class, 'unknown'));
         $this->assertSame($some, $some->filterIsOneOf([\stdClass::class, 'unknown']));
-        $this->assertSame($some, $some->filterIsOneOf(new \ArrayIterator([\stdClass::class, 'unknown'])));
         $this->assertSame($some, $some->filterIsOneOf([[\stdClass::class, 'unknown']]));
         $this->assertSame($some, $some->filterIsOneOf([[[\stdClass::class, 'unknown']]]));
         $this->assertSame($some, $some->filterIsOneOf([[[[\stdClass::class, 'unknown']]]]));
+        $this->assertSame($some, $some->filterIsOneOf(new \ArrayIterator([\stdClass::class, 'unknown'])));
+        $this->assertSame($some, $some->filterIsOneOf([
```

Comment:
is this what you were trying to do? this use case isn't supported (nested arrays of types?) perhaps I was unclear:
I would like a unit test that will fail on the current version for the bug you are attempting to fix, that runs green with your fix.

Reply:
Oh, I see. I'll add some.
The scenario is quite simple though.
You just need to create a `LazyOption` and then call `filterIsOneOf` with an array of classnames passed to it (comma-separated set of parameters works).

Reply:
Done. See `LazyOptionTest:: testFilterIsOneOf()`.
Last two assertions should be failing without my fix.